lib/bundle.js

```diff
@@ -339,14 +348,18 @@ function buildWatch(files, invalidate, build) {
       building = true;
       rebuild = false;
       
-      if (filepath) 
-        ui.log('ok', 'File `' + file + '` changed, rebuilding...');
-      else
+      if (filepath) {
+        if(relFiles.indexOf(filepath) == -1) {
+          ui.log('ok', 'File `' + file + '` changed, but not part of the bundle, skipping...');
+        } else {
+          ui.log('ok', 'File `' + file + '` changed, rebuilding...');
+        }
+      }
+      else {
         ui.log('ok', 'File changes made during previous build, rebuilding...');
-      build().then(function(newWatcherBuild) {
-        watcher.close();
-        if (rebuild)
-          newWatcherBuild();
```

Comment:
Sounds good, keep me posted and let me know if you have any questions about this rebuilding behaviour at all too.